packages/web/src/shared/components/TypeaheadInput.tsx

```diff
@@ -0,0 +1,149 @@
+import { ComponentProps, useEffect, useState } from 'react';
+import { useCombobox } from 'downshift';
+import TextInput from './TextInput';
+import { Icon } from './Icon';
+
+export interface TypeaheadInputItem {
+  label: string;
+  value: string;
+}
+
+export interface TypeaheadInputProps
+  extends Omit<ComponentProps<'input'>, 'value' | 'onChange'> {
+  value?: string;
+  items: TypeaheadInputItem[];
+  onChange(value?: string): void;
+  onCreate?(text?: string): void;
+}
+
+/**
+ * Text input with autocomplete suggestions.
+ *
+ * Suggestions have the shape of `{ label: string; value: string; }`.
+ * When an item is selected, its value will be passed to the `onChange` event.
+ *
+ * To support the creation of new items, attach the `onCreate` event.
+ * The text value will be passed to the event.
+ */
+export default function TypeaheadInput({
```

Comment:
```suggestion
export default function TypeAheadInput({
```
Are you wanting "type ahead" to be one word?

Reply:
I decided to rename it AutocompleteInput, which I think it is a lot less ambiguous

Reply:
But why don't you capitalize "Complete" to make it `AutoCompleteInput`?

Reply:
I see it as one word every where I've seen it

Reply:
😬 my bad